test/chain-test.js

```diff
@@ -761,8 +766,44 @@ describe('Chain', function() {
       'mandatory-script-verify-flag-failed');
   });
 
-  it('should cleanup', async () => {
-    await miner.close();
-    await chain.close();
+  describe('Checkpoints', function() {
+    before(async () => {
+      const entry = await chain.getEntry(chain.tip.height - 5);
+      assert(Buffer.isBuffer(entry.hash));
+      assert(Number.isInteger(entry.height));
+
+      network.checkpointMap[entry.height] = entry.hash;
+      network.lastCheckpoint = entry.height;
+    });
+
+    after(async () => {
+      network.checkpointMap = {};
+      network.lastCheckpoint = 0;
+    });
+
+    it('will reject blocks before last checkpoint', async () => {
+      const entry = await chain.getEntry(chain.tip.height - 10);
+      const block = await cpu.mineBlock(entry);
+
+      let err = null;
+
+      try {
+        await chain.add(block);
+      } catch (e) {
+        err = e;
+      }
+
+      assert(err);
+      assert.equal(err.type, 'VerifyError');
+      assert.equal(err.reason, 'bad-fork-prior-to-checkpoint');
+      assert.equal(err.score, 100);
+    });
+
+    it('will accept blocks after last checkpoint', async () => {
+      const entry = await chain.getEntry(chain.tip.height - 4);
```

Comment:
A constant that specifies the checkpoint height that is used here and when creating the checkpoint could make it more obvious that these blocks come after the last checkpoint. A comment could also suffice. This is good, just commenting on a way that I think we could make the test take less cognitive overhead to be sure of its correctness.